instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/PluginImplUtil.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v1_11;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+final class PluginImplUtil { // TODO: Copy & paste from v2
+  private PluginImplUtil() {}
+
+  private static final Logger logger = Logger.getLogger(PluginImplUtil.class.getName());
+
+  /**
+   * Check if the given {@code moduleNameImpl} is present.
+   *
+   * <p>For library instrumentations, the Impls will always be available but might fail to load if
+   * the corresponding SDK classes are not on the class path. For javaagent, the Impl is available
+   * only when the corresponding InstrumentationModule was successfully applied (muzzle passed).
+   *
+   * <p>Note that an present-but-incompatible library can only be detected by Muzzle. In
+   * library-mode, users need to ensure they are using a compatible SDK (component) versions
+   * themselves.
+   *
+   * @param implSimpleClassName The simple name of the impl class, e.g. {@code "SqsImpl"}. *
+   */
+  static boolean isImplPresent(String implSimpleClassName) {
+    // We use getName().replace() instead of getPackage() because the latter is not guaranteed to
+    // work in all cases (e.g., we might be loaded into a custom classloader that doesn't handle it)
+    String implFullClassName =
+        PluginImplUtil.class.getName().replace(".PluginImplUtil", "." + implSimpleClassName);
+    try {
+      // using package name here because library instrumentation classes are relocated when embedded
+      // in the agent
+      Class.forName(implFullClassName);
+      return true;
+    } catch (ClassNotFoundException | LinkageError e) {
+      // ClassNotFoundException will happen when muzzle disabled us in javaagent mode; LinkageError
+      // (most likely NoClassDefFoundError) should happen when the class is loaded in library mode
```

Comment:
It probably fails with `ExceptionInInitializerError` that is caused by `NoClassDefFoundError`

Reply:
I need to look into this, it would also need to be fixed for the v2 SDK. I initially added this only as additional safety net since the NoClassDefFound occured for me already without this due to the function signatures inside the class IIRC.

Reply:
Oh, actually ExceptionInInitializerError  is a LinkageError. I did the probabyl right thing without knowing... I will adapt the comment

Reply:
Cleaned up comments in c2cb01d6c6df79385ed85215ff8c88d56b758c03